src/wp-includes/comment.php

```diff
@@ -2458,6 +2458,19 @@ function wp_new_comment_notify_postauthor( $comment_id ) {
 	return wp_notify_postauthor( $comment_id );
 }
 
+/**
+ * Send a notification to the post author when a new note is added via the REST API.
+ *
+ * @since 6.9.0
+ *
+ * @param WP_Comment $comment The comment object.
+ */
+function wp_new_comment_via_rest_notify_postauthor( $comment ) {
+	if ( 'note' === $comment->comment_type ) {
+		wp_new_comment_notify_postauthor( $comment->comment_ID );
```

Comment:
Nit: PHPStan flagged this on trunk against the old PR:

<img width="1670" height="268" alt="image" src="https://github.com/user-attachments/assets/b2831955-f998-48f8-a677-59c85480efca" />


(nit because `WP_Comment::get_instance()` will cast the ID to type `int` but that's several levels deep).

```suggestion
		wp_new_comment_notify_postauthor( (int) $comment->comment_ID );
```